internal/unmarshal.go

```diff
@@ -60,6 +60,30 @@ func IsDataclass(field reflect.Value) bool {
 	return IsTypeDataclass(field.Type())
 }
 
+func IsTypeFeatureStruct(typ reflect.Type) bool {
+	if typ.Kind() == reflect.Ptr {
+		typ = typ.Elem()
+	}
+
+	if typ.Kind() != reflect.Struct {
+		// Not a dataclass nor a has-many feature.
+		return false
+	}
+
+	if typ == reflect.TypeOf(time.Time{}) {
+		return false
+	}
+
+	if IsTypeDataclass(typ) {
+		// Don't manually serialize dataclasses. Dataclasses need to be serialized
```

Comment:
This comment should be moved out but at the same time check whether the logic still stands where it should be moved.